backbone/src/main/java/org/researchstack/backbone/result/StepResult.java

```diff
@@ -139,13 +140,31 @@ public Object clone() throws CloneNotSupportedException {
         StepResult cloned = (StepResult) super.clone();
 
         cloned.results = new HashMap<>();
-        for (Map.Entry<String, T> e : results.entrySet())
-            cloned.results.put(e.getKey(), e.getValue());
+        for (Map.Entry<String, T> e : results.entrySet()) {
+            boolean isCloned = false;
+            if (e.getValue() instanceof Cloneable) {
+                try {
+                    // This is going to clone any object of the type Cloneable, This is inevitable because we need to
+                    // clone every answer in the results map in order to able to make changes to the original object
+                    // without affecting the cloned one.
+                    Method method = e.getValue().getClass().getDeclaredMethod("clone");
```

Comment:
Nasty, but I see why we need this and I cannot think of anything better at this moment.

Reply:
I spent some time thinking of a better solution but I couldn't find any :/